mlir/include/mlir/Dialect/Tensor/Utils/Utils.h

```diff
@@ -38,8 +38,14 @@ computeTransposedType(RankedTensorType rankedTensorType,
 /// i.e. for a pack from an ABCD layout to an ABCDba:
 /// The packed shape would be ABCDba.
 /// The pre-permutation shape would be AaBbCD.
-SmallVector<int64_t> getPackInverseDestPermutation(PackOp packOp);
-
+SmallVector<int64_t> getPackUnPackInverseDestPerm(
+    std::variant<tensor::PackOp, tensor::UnPackOp> packOp);
```

Comment:
I haven't seen uses of `std::variant` for similar cases in the project. Could a template work instead?